tools/logger/convert.c

```diff
@@ -1045,7 +1051,8 @@ int convert(struct convert_config *config)
 	count = fread(&snd, sizeof(snd), 1, config->ldc_fd);
 	if (!count) {
 		log_err("Error while reading %s.\n", config->ldc_file);
-		return -ferror(config->ldc_fd);
+		int result = -ferror(config->ldc_fd);
+		return result;
```

Comment:
Pointless refactor.

Reply:
It is pointless because it fixes absolutely nothing here.

Reply:
The error in this return rather comes from line 1048, `config->logs_header = &snd;`. So on return, we're left with `config->logs_header` (config is passed here as a ptr from scope higher) and local variable address (now dangling) from local scope.

Reply:
Now the issue is still weird -- why do you set a local variable to NULL? That thing can get optimized out and you _still_ get the same code. If "config" is a global variable (so that this change does become meaningful) disregard.

Reply:
Function parameters are still local variables actually.